### PR TITLE
[android] - fixup MyLocationView heading

### PR DESCRIPTION
**Pull request by tobrun**
in https://github.com/mapbox/mapbox-gl-native/pull/6660 we updated the way we get and update our camera options. This now needs to be reflected in MyLocationView code when we track the user heading.

 - Using `setCompass(0)` always shows the MyLocationView facing north when we track the user location and the heading.
 - Using ` setCompass(gpsDirection + bearing)` not only shows the direction  of the gps but recalculates it based in current heading of the map.

Review @zugaldia / @cammace

### CODE REVIEW

**Comment by zugaldia** [comment]
@cammace has been working closely with the `MyLocationView` object, I'll let him take the deal in this review.

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java [hunk 1]
```diff
@@ -702,7 +702,7 @@ void updateLatLng(@NonNull Location location) {
                     builder.bearing(location.getBearing());
                 }
                 gpsDirection = location.getBearing();
```
Comment (zugaldia) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java]: Isn't this assignment unnecessary after we change to `setCompass(0);`?

Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/widgets/MyLocationView.java]: I followed up this request with some other cleanup and corrections, want to review again?

**Comment by tobrun** [comment]
This ends up fixing multiple parts of the camera API we currenty exposed in combination with fixes for MyLocationView.

**Pull request merged**